module/os/freebsd/zfs/zio_crypt.c

```diff
@@ -1338,19 +1338,14 @@ zio_crypt_init_uios_zil(boolean_t encrypt, uint8_t *plainbuf,
 		 * authenticate it.
 		 */
 		if (txtype == TX_WRITE) {
-			crypt_len = sizeof (lr_write_t) -
-			    sizeof (lr_t) - sizeof (blkptr_t);
-			dst_iovecs[vec].iov_base = (char *)dlrp +
-			    sizeof (lr_t);
+			const size_t o = offsetof(lr_write_t, lr_blkptr);
+			crypt_len = o - sizeof (lr_t);
+			dst_iovecs[vec].iov_base = (char *)dlrp + sizeof (lr_t);
```

Comment:
The change to TX_WRITE could be a separate cleanup commit.

Reply:
I probably should add that I like the new method of calculating this. It is much easier to reason about it than the original.

Reply:
It is already quite isolated from the point of review. But if you think about minimally-invasive merges, I could possibly split it, no big deal.

Reply:
OK. Now there are two commits.